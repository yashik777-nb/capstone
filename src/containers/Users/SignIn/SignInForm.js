import React from "react";
import { withRouter } from "react-router";
import { connect } from "react-redux";
import SignIn from "../../../components/SignIn/SignIn";
import * as actionCreators from "../../../store/actions/actionsIndex";

class SingInForm extends React.Component {
  onSave(user) {
    this.props.authenticateUser(user);
    this.props.history.push("/issues");
  }

  onRegister() {
    this.props.history.push("/register");
  }

  render() {
    return (
      <SignIn
        onSave={(user) => this.onSave(user)}
        register={() => this.onRegister()}
      />
    );
  }
}

// mapDispatchToProps
const mapDispatchToProps = (dispatch) => {
  return {
    authenticateUser: (user) => dispatch(actionCreators.authenticateUser(user)),
  };
};

export default connect(null, mapDispatchToProps)(withRouter(SingInForm));
