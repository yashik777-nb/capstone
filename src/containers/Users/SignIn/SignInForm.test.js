import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createMount } from "@material-ui/core/test-utils";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";

import SignInForm from "./SignInForm";

configure({ adapter: new Adapter() });

const props = {
  authenticateUser: () => {},
};

const store = {};
const mockStore = configureStore();

describe("Sign In Form Container Tests", () => {
  let wrapper;

  beforeEach(() => {
    const render = createMount();
    wrapper = render(
      <Provider store={mockStore(store)}>
        <BrowserRouter>
          <SignInForm {...props} />
        </BrowserRouter>
      </Provider>
    );
  });

  it("Should render Sign In Form", () => {
    expect(wrapper.find("SignIn").length).toEqual(1);
  });

  test("Should Render SignIn Form Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
