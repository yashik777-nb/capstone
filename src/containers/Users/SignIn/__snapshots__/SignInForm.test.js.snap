// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Form Container Tests Should Render SignIn Form Snapshot 1`] = `ReactWrapper {}`;
